1주차 미션 / 안드로이드 1조 정일혁

1hyok opened the pull request:
## 📝 미션
- [x] 미션1
실습상에 진행한 내용들 포함하기
- [x] 미션2
프로필 카드 만들기

## 🙋 구현에 대한 설명
미션 1
실습에서 배운 Jetpack Compose 기초를 연습했습니다. 행, 열, 박스를 만들어 이미지도 삽입하고 간격, 디자인을 적절하게 조절할 수 있었습니다.
미션 2
피그마와 동일하게 프로필 카드를 구현했습니다. 앱으로 설치해 실행할 때 세로 모드일 시 공간이 작아 모양이 왜곡되었습니다. 그래서 가로로 회전해 실행 결과를 확인했습니다. 피그마가 요구하는 요소 간 간격은 각 요소마다 패딩의 크기를 지정하여 설정하였습니다.

## 📷 스크린샷 & 실행영상
![Screenshot_20250917_234438_Kuit6_week1](https://github.com/user-attachments/assets/1b4bc842-61f8-4314-8bbc-6e69cdd00f23)


## 🎸 기타

1hyok committed:
feat: Add SampleRow composable and update MainActivity

- Add SampleRow composable for arranging content horizontally.
- Update MainActivity to use SampleRow and display a black box.
- Add project configuration files for VCS, code style, and device manager.
- Update Gradle and misc XML files.

1hyok committed:
feat: Add ProfileCard composable and update MainActivity

- Add ProfileCard composable to display user profile information.
- Update MainActivity to display various UI elements like images, text, and boxes with different shapes and borders.
- Add new image resources.
- Update project configuration files for inspection profiles, deployment target, Gradle settings, app insights, and code style.

1hyok committed:
feat: Update ProfileCard composable

- Update ProfileCard to match the provided SVG design.
- Adjust dimensions, padding, and layout for better visual representation.
- Add HorizontalDivider and update text styling.
- Add Profile Card.svg.

1hyok committed:
feat: Add Practice.kt and update ProfileCard.kt

- Add Practice.kt with ModifierOrderExample and ModifierOrderExample2 composables.
- Update ProfileCard.kt to use fixed size instead of fillMaxSize for the Column.
- Update ProfileCardPreview to specify width and height.
- Update deployment target in deploymentTargetSelector.xml.

1hyok committed:
Refactor ProfileCard composable

- Updated Box and Column modifiers for explicit sizing.
- Reordered Row modifiers for clarity.
- Removed unnecessary nested Rows for Text elements.
- Enabled background in Preview.

alswlekk commented on app/src/main/java/com/example/kuit6_week1/ProfileCard.kt:
```diff
@@ -0,0 +1,232 @@
+                    shape = RoundedCornerShape(10.dp)
+                )
+                .padding(10.dp)
+        ) {
```
Column으로 처음부터 테두리 그리고 10.dp 패딩 한번에 준 거 아주 좋습니다 !

alswlekk commented on app/src/main/java/com/example/kuit6_week1/ProfileCard.kt:
```diff
@@ -0,0 +1,232 @@
+                horizontalArrangement = Arrangement.End
+            ) {
+                Image(
+                    painter = painterResource(R.drawable.image2),
```
추후에 프로젝트에 들어가서 이미지 파일이 더 많아지게 됐을 때, 이미지명을 애매하게 적어넣으면 협업을 할 때 파일 사용 시 어려움을 겪을 수가 있어서요...이미지명 이름만 봐도 어떤 파일인지 대략적으로라도 알 수 있도록 지으면 좋을 거 같습니다 !

alswlekk commented on app/src/main/java/com/example/kuit6_week1/ProfileCard.kt:
```diff
@@ -0,0 +1,232 @@
+            Text(
+                "이름 : 정일혁",
+                modifier = Modifier
+                    .width(55.dp)
```
텍스트는 이미 fontSize를 통해 크기를 정해줘서 디자이너가 무조건 이 크기에 맞게 구현해달라 !!! 하는 거 아닌 이상 따로 height나 width를 작성하지 않으셔도 됩니다 ~

alswlekk commented on app/src/main/java/com/example/kuit6_week1/ProfileCard.kt:
```diff
@@ -0,0 +1,232 @@
+                "생년월일 : 2000/05/03",
+                modifier = Modifier
+                    .width(81.dp)
+                    .padding(top = 5.dp, bottom = 5.dp),
```
Column을 써서 위에서부터 차례대로 세로로 나열하는 형식이라 만약 나머지 부분이 있으면 다 여백이 될거라 따로 bottom값은 안넘겨주셔도 됩니다 !

alswlekk reviewed (commented):
코드 작성하시느라 수고하셨어요 ~

ProtossManse commented on app/src/main/java/com/example/kuit6_week1/ProfileCard.kt:
```diff
@@ -0,0 +1,232 @@
+            modifier = Modifier
+//                .width(240.dp)
+//                .size(width = 140.dp, height = 240.dp)
+                .size(width = 240.dp, height = 140.dp)
```
이 부분 height를 지정해주지 않고, preview의 height를 늘리면 저번에 말씀하셨던 글자가 잘리는 문제가 해결됩니다
폰트가 다르기도 하고, 또 Column 안에 패딩이 있기 때문에 피그마와 똑같이 높이를 140으로 설정하더라도 동일하게 보이지 않을 수 있어요!

ProtossManse reviewed (approved):
미션하느라 수고하셨습니다~

1hyok committed:
feat: Add ProfileCard2 composable and update MainActivity

- Add ProfileCard2 composable to display user profile details including name, student ID, department, and birthdate.
- Update MainActivity to use ProfileCard2 as the main content.
- Add `kuit.png` image resource.
- Update `libs.versions.toml` and `build.gradle.kts` to include the Compose foundation-layout dependency.
